arrows/klv/klv_data_format.cxx

```diff
@@ -561,6 +795,34 @@ ::description() const
   return ss.str();
 }
 
+#define KLV_INSTANTIATE( T ) \
```

Comment:
This is not critical, and need not hold up merging, but I have a suggestion on how to organize this code so that all the instantiations of all the types are not all in this one file.  It's a pattern that is used commonly in VXL.  You make a template header file that contains all the template function definitions that you've added in this file and the instantiate macros.  This file often gets a special extension, like `.hxx`, and gets include guards like a header file, but it's not a public header.  We used to use `.txx`, but switched to `.hxx` because it is more commonly recognized as C++ by IDEs.  For each type you want to instantiate you create a `.cxx` file for that type which would look something like this:

File name: `klv_data_format+klv_blob-.cxx`
Contents:
```c++
#include "klv_blob.h"
#include "klv_data_format.hxx"

KLV_INSTANTIATE( klv_blob );
```
In VXL, all of these special instantiation files go in a `templates` subdirectory.  The advantage of this approach is that you keep the template implementations separate from your instantiation for all the various types.  This allows you to compile them in parallel and you don't have one massive file that has to include everything.